// @nebular/theme/esm2020/components/button/button.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbButtonComponent } from './button.component';
import * as i0 from "@angular/core";
const NB_BUTTON_COMPONENTS = [
    NbButtonComponent,
];
export class NbButtonModule {
}
NbButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbButtonModule, declarations: [NbButtonComponent], imports: [NbSharedModule], exports: [NbButtonComponent] });
NbButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbButtonModule, imports: [[
            NbSharedModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        NbSharedModule,
                    ],
                    declarations: [
                        ...NB_BUTTON_COMPONENTS,
                    ],
                    exports: [
                        ...NB_BUTTON_COMPONENTS,
                    ],
                }]
        }] });
