// @nebular/theme/esm2020/components/cdk/overlay/overlay-service.mjs
import { Injectable } from '@angular/core';
import { NbComponentPortal, } from './mapping';
import * as i0 from "@angular/core";
import * as i1 from "./mapping";
import * as i2 from "../../../services/direction.service";
export function patch(container, containerContext) {
    Object.assign(container.instance, containerContext);
    container.changeDetectorRef.detectChanges();
    return container;
}
export function createContainer(ref, container, context, componentFactoryResolver) {
    const containerRef = ref.attach(new NbComponentPortal(container, null, null, componentFactoryResolver));
    patch(containerRef, context);
    return containerRef;
}
export class NbOverlayService {
    constructor(overlay, layoutDirection) {
        this.overlay = overlay;
        this.layoutDirection = layoutDirection;
    }
    get scrollStrategies() {
        return this.overlay.scrollStrategies;
    }
    create(config) {
        const overlayRef = this.overlay.create(config);
        this.layoutDirection.onDirectionChange()
            .subscribe(dir => overlayRef.setDirection(dir));
        return overlayRef;
    }
}
NbOverlayService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOverlayService, deps: [{ token: i1.NbOverlay }, { token: i2.NbLayoutDirectionService }], target: i0.ɵɵFactoryTarget.Injectable });
NbOverlayService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOverlayService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOverlayService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.NbOverlay }, { type: i2.NbLayoutDirectionService }]; } });
