// @nebular/theme/esm2020/components/autocomplete/autocomplete.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbOverlayModule } from '../cdk/overlay/overlay.module';
import { NbCardModule } from '../card/card.module';
import { NbAutocompleteComponent } from './autocomplete.component';
import { NbAutocompleteDirective } from './autocomplete.directive';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { NbOptionModule } from '../option/option-list.module';
import * as i0 from "@angular/core";
const NB_AUTOCOMPLETE_COMPONENTS = [
    NbAutocompleteComponent,
    NbAutocompleteDirective,
];
export class NbAutocompleteModule {
}
NbAutocompleteModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAutocompleteModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbAutocompleteModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAutocompleteModule, declarations: [NbAutocompleteComponent,
        NbAutocompleteDirective], imports: [CommonModule,
        FormsModule,
        NbOverlayModule,
        NbCardModule,
        NbOptionModule], exports: [NbAutocompleteComponent,
        NbAutocompleteDirective, NbOptionModule] });
NbAutocompleteModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAutocompleteModule, imports: [[
            CommonModule,
            FormsModule,
            NbOverlayModule,
            NbCardModule,
            NbOptionModule,
        ], NbOptionModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAutocompleteModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        NbOverlayModule,
                        NbCardModule,
                        NbOptionModule,
                    ],
                    exports: [
                        ...NB_AUTOCOMPLETE_COMPONENTS,
                        NbOptionModule,
                    ],
                    declarations: [...NB_AUTOCOMPLETE_COMPONENTS],
                }]
        }] });
