// @nebular/theme/esm2020/components/card/shared/shared.component.mjs
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Component intended to be used within the `<nb-flip-card>` and `<nb-reveal-card>` components.
 *
 * Use it as a container for the front card.
 */
export class NbCardFrontComponent {
}
NbCardFrontComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbCardFrontComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
NbCardFrontComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbCardFrontComponent, selector: "nb-card-front", ngImport: i0, template: '<ng-content select="nb-card"></ng-content>', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbCardFrontComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-card-front',
                    template: '<ng-content select="nb-card"></ng-content>',
                }]
        }] });
/**
 * Component intended to be used within the `<nb-flip-card>` and `<nb-reveal-card>` components.
 *
 * Use it as a container for the back card.
 */
export class NbCardBackComponent {
}
NbCardBackComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbCardBackComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
NbCardBackComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbCardBackComponent, selector: "nb-card-back", ngImport: i0, template: '<ng-content select="nb-card"></ng-content>', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbCardBackComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-card-back',
                    template: '<ng-content select="nb-card"></ng-content>',
                }]
        }] });
