// @nebular/theme/esm2020/components/timepicker/timepicker-cell.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostListener, Input, Output, ViewChild, } from '@angular/core';
import { filter, take, takeUntil } from 'rxjs/operators';
import { merge, Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../cdk/platform/platform-service";
export class NbTimePickerCellComponent {
    constructor(ngZone, platformService) {
        this.ngZone = ngZone;
        this.platformService = platformService;
        this.selectedChange$ = new Subject();
        this.unselected$ = this.selectedChange$.pipe(filter((selected) => !selected));
        this.destroy$ = new Subject();
        this.select = new EventEmitter();
    }
    set selected(selected) {
        if (selected) {
            this._selected = selected;
            this.scrollToElement();
        }
        this.selectedChange$.next(selected);
    }
    ;
    get selected() {
        return this._selected;
    }
    onClick() {
        this.select.emit({ value: this.value });
    }
    ngAfterViewInit() {
        if (this.selected) {
            // Since we render timepicker in the overlay, at the moment this hook called,
            // timepicker could be not fully rendered and placed. Because of it, we're waiting for Angular
            // to finish change detection run and only then scroll to the selected cell.
            this.ngZone.onStable
                .pipe(take(1), takeUntil(merge(this.unselected$, this.destroy$)))
                .subscribe(() => this.scrollToElement());
        }
    }
    scrollToElement() {
        if (this.valueContainerElement && this.platformService.isBrowser) {
            this.ngZone.runOutsideAngular(() => this.valueContainerElement.nativeElement.scrollIntoView({ block: 'center' }));
        }
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
}
NbTimePickerCellComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTimePickerCellComponent, deps: [{ token: i0.NgZone }, { token: i1.NbPlatform }], target: i0.ɵɵFactoryTarget.Component });
NbTimePickerCellComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbTimePickerCellComponent, selector: "nb-timepicker-cell", inputs: { selected: "selected", value: "value" }, outputs: { select: "select" }, host: { listeners: { "click": "onClick()" } }, viewQueries: [{ propertyName: "valueContainerElement", first: true, predicate: ["valueContainer"], descendants: true }], ngImport: i0, template: `
    <div #valueContainer>{{ value }}</div>
  `, isInline: true, styles: ["/**\n * @license\n * Copyright Akveo. All Rights Reserved.\n * Licensed under the MIT License. See License.txt in the project root for license information.\n */:host{width:100%;height:100%;display:flex;align-items:center;justify-content:center;-webkit-user-select:none;-ms-user-select:none;user-select:none}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTimePickerCellComponent, decorators: [{
            type: Component,
            args: [{ selector: 'nb-timepicker-cell', template: `
    <div #valueContainer>{{ value }}</div>
  `, changeDetection: ChangeDetectionStrategy.OnPush, styles: ["/**\n * @license\n * Copyright Akveo. All Rights Reserved.\n * Licensed under the MIT License. See License.txt in the project root for license information.\n */:host{width:100%;height:100%;display:flex;align-items:center;justify-content:center;-webkit-user-select:none;-ms-user-select:none;user-select:none}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.NgZone }, { type: i1.NbPlatform }]; }, propDecorators: { selected: [{
                type: Input
            }], value: [{
                type: Input
            }], select: [{
                type: Output
            }], valueContainerElement: [{
                type: ViewChild,
                args: ['valueContainer']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
