// @nebular/theme/esm2020/components/timepicker/timepicker.module.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NbOverlayModule } from '../cdk/overlay/overlay.module';
import { NbListModule } from '../list/list.module';
import { NbCardModule } from '../card/card.module';
import { NbCalendarKitModule } from '../calendar-kit/calendar-kit.module';
import { NbTimePickerDirective } from './timepicker.directive';
import { NbTimePickerComponent } from './timepicker.component';
import { NbTimePickerCellComponent } from './timepicker-cell.component';
import { NbCalendarTimeModelService } from '../calendar-kit/services/calendar-time-model.service';
import { NB_TIME_PICKER_CONFIG } from './model';
import * as i0 from "@angular/core";
export class NbTimepickerModule {
    static forRoot(config = {}) {
        return {
            ngModule: NbTimepickerModule,
            providers: [{ provide: NB_TIME_PICKER_CONFIG, useValue: config }],
        };
    }
    static forChild(config = {}) {
        return {
            ngModule: NbTimepickerModule,
            providers: [{ provide: NB_TIME_PICKER_CONFIG, useValue: config }],
        };
    }
}
NbTimepickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTimepickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbTimepickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTimepickerModule, declarations: [NbTimePickerComponent, NbTimePickerCellComponent, NbTimePickerDirective], imports: [CommonModule,
        NbOverlayModule,
        NbListModule,
        NbCardModule,
        NbCalendarKitModule], exports: [NbTimePickerComponent, NbTimePickerCellComponent, NbTimePickerDirective] });
NbTimepickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTimepickerModule, providers: [NbCalendarTimeModelService], imports: [[
            CommonModule,
            NbOverlayModule,
            NbListModule,
            NbCardModule,
            NbCalendarKitModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTimepickerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        NbOverlayModule,
                        NbListModule,
                        NbCardModule,
                        NbCalendarKitModule,
                    ],
                    providers: [NbCalendarTimeModelService],
                    exports: [NbTimePickerComponent, NbTimePickerCellComponent, NbTimePickerDirective],
                    declarations: [NbTimePickerComponent, NbTimePickerCellComponent, NbTimePickerDirective],
                }]
        }] });
