// @nebular/theme/esm2020/components/context-menu/context-menu.component.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Component, Input } from '@angular/core';
import { NbPositionedContainerComponent } from '../cdk/overlay/overlay-container';
import * as i0 from "@angular/core";
import * as i1 from "../menu/menu.component";
/**
 * Context menu component used as content within NbContextMenuDirective.
 *
 * @styles
 *
 * context-menu-background-color:
 * context-menu-border-color:
 * context-menu-border-style:
 * context-menu-border-width:
 * context-menu-border-radius:
 * context-menu-text-align:
 * context-menu-min-width:
 * context-menu-max-width:
 * context-menu-shadow:
 * */
export class NbContextMenuComponent extends NbPositionedContainerComponent {
    constructor() {
        super(...arguments);
        this.items = [];
        this.context = { items: [] };
    }
    /**
     * The method is empty since we don't need to do anything additionally
     * render is handled by change detection
     */
    renderContent() { }
}
NbContextMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbContextMenuComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NbContextMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbContextMenuComponent, selector: "nb-context-menu", inputs: { items: "items", tag: "tag", context: "context" }, usesInheritance: true, ngImport: i0, template: `
    <nb-menu class="context-menu" [items]="context.items" [tag]="context.tag"></nb-menu>
  `, isInline: true, components: [{ type: i1.NbMenuComponent, selector: "nb-menu", inputs: ["tag", "items", "autoCollapse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbContextMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-context-menu',
                    template: `
    <nb-menu class="context-menu" [items]="context.items" [tag]="context.tag"></nb-menu>
  `,
                }]
        }], propDecorators: { items: [{
                type: Input
            }], tag: [{
                type: Input
            }], context: [{
                type: Input
            }] } });
