// @nebular/theme/esm2020/components/toggle/toggle.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NbIconModule } from '../icon/icon.module';
import { NbToggleComponent } from './toggle.component';
import * as i0 from "@angular/core";
export class NbToggleModule {
}
NbToggleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbToggleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbToggleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbToggleModule, declarations: [NbToggleComponent], imports: [CommonModule,
        NbIconModule], exports: [NbToggleComponent] });
NbToggleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbToggleModule, imports: [[
            CommonModule,
            NbIconModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbToggleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        NbIconModule,
                    ],
                    declarations: [NbToggleComponent],
                    exports: [NbToggleComponent],
                }]
        }] });
