// @nebular/theme/esm2020/components/datepicker/datepicker-container.component.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Component, ViewChild } from '@angular/core';
import { NbOverlayContainerComponent, NbPositionedContainerComponent } from '../cdk/overlay/overlay-container';
import * as i0 from "@angular/core";
import * as i1 from "../cdk/overlay/overlay-container";
export class NbDatepickerContainerComponent extends NbPositionedContainerComponent {
    attach(portal) {
        return this.overlayContainer.attachComponentPortal(portal);
    }
}
NbDatepickerContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbDatepickerContainerComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NbDatepickerContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbDatepickerContainerComponent, selector: "nb-datepicker-container", viewQueries: [{ propertyName: "overlayContainer", first: true, predicate: NbOverlayContainerComponent, descendants: true, static: true }], usesInheritance: true, ngImport: i0, template: `
    <nb-overlay-container></nb-overlay-container>
  `, isInline: true, components: [{ type: i1.NbOverlayContainerComponent, selector: "nb-overlay-container" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbDatepickerContainerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-datepicker-container',
                    template: `
    <nb-overlay-container></nb-overlay-container>
  `,
                }]
        }], propDecorators: { overlayContainer: [{
                type: ViewChild,
                args: [NbOverlayContainerComponent, { static: true }]
            }] } });
