// @nebular/theme/esm2020/components/cdk/table/row.mjs
import { Component, Directive, Input } from '@angular/core';
import { CdkFooterRow, CdkFooterRowDef, CdkHeaderRow, CdkHeaderRowDef, CdkRow, CdkRowDef, CdkCellOutlet, DataRowOutlet, HeaderRowOutlet, FooterRowOutlet, NoDataRowOutlet, } from '@angular/cdk/table';
import * as i0 from "@angular/core";
export class NbDataRowOutletDirective extends DataRowOutlet {
}
NbDataRowOutletDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbDataRowOutletDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbDataRowOutletDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbDataRowOutletDirective, selector: "[nbRowOutlet]", providers: [{ provide: DataRowOutlet, useExisting: NbDataRowOutletDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbDataRowOutletDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbRowOutlet]',
                    providers: [{ provide: DataRowOutlet, useExisting: NbDataRowOutletDirective }],
                }]
        }] });
export class NbHeaderRowOutletDirective extends HeaderRowOutlet {
}
NbHeaderRowOutletDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbHeaderRowOutletDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbHeaderRowOutletDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbHeaderRowOutletDirective, selector: "[nbHeaderRowOutlet]", providers: [{ provide: HeaderRowOutlet, useExisting: NbHeaderRowOutletDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbHeaderRowOutletDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbHeaderRowOutlet]',
                    providers: [{ provide: HeaderRowOutlet, useExisting: NbHeaderRowOutletDirective }],
                }]
        }] });
export class NbFooterRowOutletDirective extends FooterRowOutlet {
}
NbFooterRowOutletDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFooterRowOutletDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbFooterRowOutletDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbFooterRowOutletDirective, selector: "[nbFooterRowOutlet]", providers: [{ provide: FooterRowOutlet, useExisting: NbFooterRowOutletDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFooterRowOutletDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbFooterRowOutlet]',
                    providers: [{ provide: FooterRowOutlet, useExisting: NbFooterRowOutletDirective }],
                }]
        }] });
export class NbNoDataRowOutletDirective extends NoDataRowOutlet {
}
NbNoDataRowOutletDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbNoDataRowOutletDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbNoDataRowOutletDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbNoDataRowOutletDirective, selector: "[nbNoDataRowOutlet]", providers: [{ provide: NoDataRowOutlet, useExisting: NbNoDataRowOutletDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbNoDataRowOutletDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbNoDataRowOutlet]',
                    providers: [{ provide: NoDataRowOutlet, useExisting: NbNoDataRowOutletDirective }],
                }]
        }] });
export class NbCellOutletDirective extends CdkCellOutlet {
}
NbCellOutletDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbCellOutletDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbCellOutletDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbCellOutletDirective, selector: "[nbCellOutlet]", providers: [{ provide: CdkCellOutlet, useExisting: NbCellOutletDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbCellOutletDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbCellOutlet]',
                    providers: [{ provide: CdkCellOutlet, useExisting: NbCellOutletDirective }],
                }]
        }] });
/**
 * Header row definition for the nb-table.
 * Captures the header row's template and other header properties such as the columns to display.
 */
export class NbHeaderRowDefDirective extends CdkHeaderRowDef {
}
NbHeaderRowDefDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbHeaderRowDefDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbHeaderRowDefDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbHeaderRowDefDirective, selector: "[nbHeaderRowDef]", inputs: { columns: ["nbHeaderRowDef", "columns"], sticky: ["nbHeaderRowDefSticky", "sticky"] }, providers: [{ provide: CdkHeaderRowDef, useExisting: NbHeaderRowDefDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbHeaderRowDefDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbHeaderRowDef]',
                    providers: [{ provide: CdkHeaderRowDef, useExisting: NbHeaderRowDefDirective }],
                }]
        }], propDecorators: { columns: [{
                type: Input,
                args: ['nbHeaderRowDef']
            }], sticky: [{
                type: Input,
                args: ['nbHeaderRowDefSticky']
            }] } });
/**
 * Footer row definition for the nb-table.
 * Captures the footer row's template and other footer properties such as the columns to display.
 */
export class NbFooterRowDefDirective extends CdkFooterRowDef {
}
NbFooterRowDefDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFooterRowDefDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbFooterRowDefDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbFooterRowDefDirective, selector: "[nbFooterRowDef]", inputs: { columns: ["nbFooterRowDef", "columns"], sticky: ["nbFooterRowDefSticky", "sticky"] }, providers: [{ provide: CdkFooterRowDef, useExisting: NbFooterRowDefDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFooterRowDefDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbFooterRowDef]',
                    providers: [{ provide: CdkFooterRowDef, useExisting: NbFooterRowDefDirective }],
                }]
        }], propDecorators: { columns: [{
                type: Input,
                args: ['nbFooterRowDef']
            }], sticky: [{
                type: Input,
                args: ['nbFooterRowDefSticky']
            }] } });
/**
 * Data row definition for the nb-table.
 * Captures the data row's template and other properties such as the columns to display and
 * a when predicate that describes when this row should be used.
 */
export class NbRowDefDirective extends CdkRowDef {
}
NbRowDefDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRowDefDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbRowDefDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbRowDefDirective, selector: "[nbRowDef]", inputs: { columns: ["nbRowDefColumns", "columns"], when: ["nbRowDefWhen", "when"] }, providers: [{ provide: CdkRowDef, useExisting: NbRowDefDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRowDefDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbRowDef]',
                    providers: [{ provide: CdkRowDef, useExisting: NbRowDefDirective }],
                }]
        }], propDecorators: { columns: [{
                type: Input,
                args: ['nbRowDefColumns']
            }], when: [{
                type: Input,
                args: ['nbRowDefWhen']
            }] } });
/** Footer template container that contains the cell outlet. Adds the right class and role. */
export class NbHeaderRowComponent extends CdkHeaderRow {
}
NbHeaderRowComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbHeaderRowComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NbHeaderRowComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbHeaderRowComponent, selector: "nb-header-row, tr[nbHeaderRow]", host: { attributes: { "role": "row" }, classAttribute: "nb-header-row" }, providers: [{ provide: CdkHeaderRow, useExisting: NbHeaderRowComponent }], usesInheritance: true, ngImport: i0, template: `
    <ng-container nbCellOutlet></ng-container>`, isInline: true, directives: [{ type: NbCellOutletDirective, selector: "[nbCellOutlet]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbHeaderRowComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-header-row, tr[nbHeaderRow]',
                    template: `
    <ng-container nbCellOutlet></ng-container>`,
                    host: {
                        'class': 'nb-header-row',
                        'role': 'row',
                    },
                    providers: [{ provide: CdkHeaderRow, useExisting: NbHeaderRowComponent }],
                }]
        }] });
/** Footer template container that contains the cell outlet. Adds the right class and role. */
export class NbFooterRowComponent extends CdkFooterRow {
}
NbFooterRowComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFooterRowComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NbFooterRowComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbFooterRowComponent, selector: "nb-footer-row, tr[nbFooterRow]", host: { attributes: { "role": "row" }, classAttribute: "nb-footer-row" }, providers: [{ provide: CdkFooterRow, useExisting: NbFooterRowComponent }], usesInheritance: true, ngImport: i0, template: `
    <ng-container nbCellOutlet></ng-container>`, isInline: true, directives: [{ type: NbCellOutletDirective, selector: "[nbCellOutlet]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFooterRowComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-footer-row, tr[nbFooterRow]',
                    template: `
    <ng-container nbCellOutlet></ng-container>`,
                    host: {
                        'class': 'nb-footer-row',
                        'role': 'row',
                    },
                    providers: [{ provide: CdkFooterRow, useExisting: NbFooterRowComponent }],
                }]
        }] });
/** Data row template container that contains the cell outlet. Adds the right class and role. */
export class NbRowComponent extends CdkRow {
}
NbRowComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRowComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
NbRowComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbRowComponent, selector: "nb-row, tr[nbRow]", host: { attributes: { "role": "row" }, classAttribute: "nb-row" }, providers: [{ provide: CdkRow, useExisting: NbRowComponent }], usesInheritance: true, ngImport: i0, template: `
    <ng-container nbCellOutlet></ng-container>`, isInline: true, directives: [{ type: NbCellOutletDirective, selector: "[nbCellOutlet]" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRowComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-row, tr[nbRow]',
                    template: `
    <ng-container nbCellOutlet></ng-container>`,
                    host: {
                        'class': 'nb-row',
                        'role': 'row',
                    },
                    providers: [{ provide: CdkRow, useExisting: NbRowComponent }],
                }]
        }] });
