// @nebular/theme/esm2020/components/option/option-list.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NbOptionComponent } from './option.component';
import { NbOptionGroupComponent } from './option-group.component';
import { NbOptionListComponent } from './option-list.component';
import { NbCheckboxModule } from '../checkbox/checkbox.module';
import * as i0 from "@angular/core";
const NB_OPTION_LIST_COMPONENTS = [
    NbOptionListComponent,
    NbOptionComponent,
    NbOptionGroupComponent,
];
export class NbOptionModule {
}
NbOptionModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOptionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbOptionModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOptionModule, declarations: [NbOptionListComponent,
        NbOptionComponent,
        NbOptionGroupComponent], imports: [CommonModule,
        NbCheckboxModule], exports: [NbOptionListComponent,
        NbOptionComponent,
        NbOptionGroupComponent] });
NbOptionModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOptionModule, imports: [[
            CommonModule,
            NbCheckboxModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbOptionModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ...NB_OPTION_LIST_COMPONENTS,
                    ],
                    imports: [
                        CommonModule,
                        NbCheckboxModule,
                    ],
                    exports: [
                        ...NB_OPTION_LIST_COMPONENTS,
                    ],
                }]
        }] });
