// @nebular/theme/esm2020/components/user/user.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbUserComponent, } from './user.component';
import { NbBadgeModule } from '../badge/badge.module';
import * as i0 from "@angular/core";
const NB_USER_COMPONENTS = [
    NbUserComponent,
];
export class NbUserModule {
}
NbUserModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbUserModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbUserModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbUserModule, declarations: [NbUserComponent], imports: [NbSharedModule,
        NbBadgeModule], exports: [NbUserComponent] });
NbUserModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbUserModule, imports: [[
            NbSharedModule,
            NbBadgeModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbUserModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        NbSharedModule,
                        NbBadgeModule,
                    ],
                    declarations: [
                        ...NB_USER_COMPONENTS,
                    ],
                    exports: [
                        ...NB_USER_COMPONENTS,
                    ],
                }]
        }] });
