// @nebular/theme/esm2020/components/progress-bar/progress-bar.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbProgressBarComponent } from './progress-bar.component';
import * as i0 from "@angular/core";
export class NbProgressBarModule {
}
NbProgressBarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbProgressBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbProgressBarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbProgressBarModule, declarations: [NbProgressBarComponent], imports: [NbSharedModule], exports: [NbProgressBarComponent] });
NbProgressBarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbProgressBarModule, imports: [[
            NbSharedModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbProgressBarModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        NbSharedModule,
                    ],
                    declarations: [NbProgressBarComponent],
                    exports: [NbProgressBarComponent],
                }]
        }] });
