// @nebular/theme/esm2020/components/list/list.component.mjs
import { Component, Input, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * List is a container component that wraps `nb-list-item` component.
 *
 * Basic example:
 * @stacked-example(Simple list, list/simple-list-showcase.component)
 *
 * `nb-list-item` accepts arbitrary content, so you can create a list of any components.
 *
 * ### Installation
 *
 * Import `NbListModule` to your feature module.
 * ```ts
 * @NgModule({
 *   imports: [
 *     // ...
 *     NbListModule,
 *   ],
 * })
 * export class PageModule { }
 * ```
 * ### Usage
 *
 * List of users:
 * @stacked-example(Users list, list/users-list-showcase.component)
 *
 * @styles
 *
 * list-item-divider-color:
 * list-item-divider-style:
 * list-item-divider-width:
 * list-item-padding:
 * list-item-text-color:
 * list-item-font-family:
 * list-item-font-size:
 * list-item-font-weight:
 * list-item-line-height:
 */
export class NbListComponent {
    constructor() {
        /**
         * Role attribute value
         *
         * @type {string}
         */
        this.role = 'list';
    }
}
NbListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
NbListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbListComponent, selector: "nb-list", inputs: { role: "role" }, host: { properties: { "attr.role": "this.role" } }, ngImport: i0, template: `<ng-content select="nb-list-item"></ng-content>`, isInline: true, styles: [":host{display:flex;flex-direction:column;flex:1 1 auto;overflow:auto}\n"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbListComponent, decorators: [{
            type: Component,
            args: [{ selector: 'nb-list', template: `<ng-content select="nb-list-item"></ng-content>`, styles: [":host{display:flex;flex-direction:column;flex:1 1 auto;overflow:auto}\n"] }]
        }], propDecorators: { role: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['attr.role']
            }] } });
/**
 * List item component is a grouping component that accepts arbitrary content.
 * It should be direct child of `nb-list` componet.
 */
export class NbListItemComponent {
    constructor() {
        /**
         * Role attribute value
         *
         * @type {string}
         */
        this.role = 'listitem';
    }
}
NbListItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbListItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
NbListItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbListItemComponent, selector: "nb-list-item", inputs: { role: "role" }, host: { properties: { "attr.role": "this.role" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, styles: [":host{display:flex;align-items:center;flex-shrink:0}\n"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbListItemComponent, decorators: [{
            type: Component,
            args: [{ selector: 'nb-list-item', template: `<ng-content></ng-content>`, styles: [":host{display:flex;align-items:center;flex-shrink:0}\n"] }]
        }], propDecorators: { role: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['attr.role']
            }] } });
