// @nebular/theme/esm2020/components/tabset/tabset.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbTabsetComponent, NbTabComponent } from './tabset.component';
import { NbBadgeModule } from '../badge/badge.module';
import { NbIconModule } from '../icon/icon.module';
import { NbTabContentDirective } from './tab-content.directive';
import { NbTabTitleDirective } from './tab-title.directive';
import * as i0 from "@angular/core";
const NB_TABSET_COMPONENTS = [NbTabsetComponent, NbTabComponent];
const NB_TABSET_DIRECTIVES = [NbTabContentDirective, NbTabTitleDirective];
export class NbTabsetModule {
}
NbTabsetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTabsetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbTabsetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTabsetModule, declarations: [NbTabsetComponent, NbTabComponent, NbTabContentDirective, NbTabTitleDirective], imports: [NbSharedModule, NbBadgeModule, NbIconModule], exports: [NbTabsetComponent, NbTabComponent, NbTabContentDirective, NbTabTitleDirective] });
NbTabsetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTabsetModule, imports: [[NbSharedModule, NbBadgeModule, NbIconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTabsetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [NbSharedModule, NbBadgeModule, NbIconModule],
                    declarations: [...NB_TABSET_COMPONENTS, ...NB_TABSET_DIRECTIVES],
                    exports: [...NB_TABSET_COMPONENTS, ...NB_TABSET_DIRECTIVES],
                }]
        }] });
