// @nebular/theme/esm2020/components/tree-grid/tree-grid-row-toggle.directive.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Directive, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./tree-grid-cell.component";
/**
 * When using custom row toggle, apply this directive on your toggle to toggle row on element click.
 */
export class NbTreeGridRowToggleDirective {
    constructor(cell) {
        this.cell = cell;
    }
    toggleRow($event) {
        this.cell.toggleRow();
        $event.stopPropagation();
    }
}
NbTreeGridRowToggleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridRowToggleDirective, deps: [{ token: i1.NbTreeGridCellDirective }], target: i0.ɵɵFactoryTarget.Directive });
NbTreeGridRowToggleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbTreeGridRowToggleDirective, selector: "[nbTreeGridRowToggle]", host: { listeners: { "click": "toggleRow($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridRowToggleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbTreeGridRowToggle]',
                }]
        }], ctorParameters: function () { return [{ type: i1.NbTreeGridCellDirective }]; }, propDecorators: { toggleRow: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
