// @nebular/theme/esm2020/components/tree-grid/tree-grid-row-toggle.component.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Component, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./tree-grid-cell.component";
import * as i2 from "../icon/icon.component";
/**
 * NbTreeGridRowToggleComponent
 */
export class NbTreeGridRowToggleComponent {
    constructor(cell) {
        this.cell = cell;
    }
    set expanded(value) {
        this.expandedValue = value;
    }
    get expanded() {
        return this.expandedValue;
    }
    toggleRow($event) {
        this.cell.toggleRow();
        $event.stopPropagation();
    }
}
NbTreeGridRowToggleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridRowToggleComponent, deps: [{ token: i1.NbTreeGridCellDirective }], target: i0.ɵɵFactoryTarget.Component });
NbTreeGridRowToggleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.0", type: NbTreeGridRowToggleComponent, selector: "nb-tree-grid-row-toggle", inputs: { expanded: "expanded" }, host: { listeners: { "click": "toggleRow($event)" } }, ngImport: i0, template: `
    <button class="row-toggle-button" [attr.aria-label]="expanded ? 'collapse' : 'expand'">
      <nb-icon [icon]="expanded ? 'chevron-down-outline' : 'chevron-right-outline'"
               pack="nebular-essentials"
               aria-hidden="true">
      </nb-icon>
    </button>
  `, isInline: true, styles: ["button{background:transparent;border:none;padding:0}\n"], components: [{ type: i2.NbIconComponent, selector: "nb-icon", inputs: ["icon", "pack", "options", "status", "config"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridRowToggleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'nb-tree-grid-row-toggle',
                    template: `
    <button class="row-toggle-button" [attr.aria-label]="expanded ? 'collapse' : 'expand'">
      <nb-icon [icon]="expanded ? 'chevron-down-outline' : 'chevron-right-outline'"
               pack="nebular-essentials"
               aria-hidden="true">
      </nb-icon>
    </button>
  `,
                    styles: [`
    button {
      background: transparent;
      border: none;
      padding: 0;
    }
  `],
                }]
        }], ctorParameters: function () { return [{ type: i1.NbTreeGridCellDirective }]; }, propDecorators: { expanded: [{
                type: Input
            }], toggleRow: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
