// @nebular/theme/esm2020/components/tree-grid/tree-grid-cell.component.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Directive, HostBinding, Inject, PLATFORM_ID, } from '@angular/core';
import { isPlatformBrowser } from '@angular/common';
import { filter, takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import { NB_WINDOW } from '../../theme.options';
import { NbCellDirective, NbFooterCellDirective, NbHeaderCellDirective } from '../cdk/table/cell';
import { NbCdkCell, NbCdkFooterCell, NbCdkHeaderCell } from '../cdk/table/type-mappings';
import { NB_TREE_GRID } from './tree-grid-injection-tokens';
import { NB_DEFAULT_ROW_LEVEL } from './data-source/tree-grid.model';
import * as i0 from "@angular/core";
import * as i1 from "./tree-grid-column-def.directive";
import * as i2 from "@angular/platform-browser";
import * as i3 from "../../services/direction.service";
import * as i4 from "./tree-grid-columns.service";
export class NbTreeGridCellDirective extends NbCellDirective {
    constructor(columnDef, elementRef, tree, platformId, window, sanitizer, directionService, columnService, cd) {
        super(columnDef, elementRef);
        this.platformId = platformId;
        this.window = window;
        this.sanitizer = sanitizer;
        this.directionService = directionService;
        this.columnService = columnService;
        this.cd = cd;
        this.destroy$ = new Subject();
        this.initialLeftPadding = '';
        this.initialRightPadding = '';
        this.tree = tree;
        this.columnDef = columnDef;
        this.elementRef = elementRef;
    }
    get columnWidth() {
        this.latestWidth = this.tree.getColumnWidth();
        if (this.latestWidth) {
            return this.latestWidth;
        }
        return null;
    }
    get leftPadding() {
        if (this.directionService.isLtr()) {
            return this.getStartPadding();
        }
        return null;
    }
    get rightPadding() {
        if (this.directionService.isRtl()) {
            return this.getStartPadding();
        }
        return null;
    }
    ngOnInit() {
        if (isPlatformBrowser(this.platformId)) {
            const style = this.window.getComputedStyle(this.elementRef.nativeElement);
            this.initialLeftPadding = style.paddingLeft;
            this.initialRightPadding = style.paddingRight;
        }
        this.columnService.onColumnsChange()
            .pipe(filter(() => this.latestWidth !== this.tree.getColumnWidth()), takeUntil(this.destroy$))
            .subscribe(() => this.cd.detectChanges());
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    toggleRow() {
        this.tree.toggleCellRow(this);
    }
    get initialStartPadding() {
        return this.directionService.isLtr()
            ? this.initialLeftPadding
            : this.initialRightPadding;
    }
    getStartPadding() {
        const rowLevel = this.tree.getCellLevel(this, this.columnDef.name);
        if (rowLevel === NB_DEFAULT_ROW_LEVEL) {
            return null;
        }
        const nestingLevel = rowLevel + 1;
        let padding = '';
        if (this.tree.levelPadding) {
            padding = `calc(${this.tree.levelPadding} * ${nestingLevel})`;
        }
        else if (this.initialStartPadding) {
            padding = `calc(${this.initialStartPadding} * ${nestingLevel})`;
        }
        if (!padding) {
            return null;
        }
        return this.sanitizer.bypassSecurityTrustStyle(padding);
    }
}
NbTreeGridCellDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridCellDirective, deps: [{ token: i1.NbTreeGridColumnDefDirective }, { token: i0.ElementRef }, { token: NB_TREE_GRID }, { token: PLATFORM_ID }, { token: NB_WINDOW }, { token: i2.DomSanitizer }, { token: i3.NbLayoutDirectionService }, { token: i4.NbColumnsService }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Directive });
NbTreeGridCellDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbTreeGridCellDirective, selector: "td[nbTreeGridCell]", host: { attributes: { "role": "gridcell" }, properties: { "style.width": "this.columnWidth", "style.padding-left": "this.leftPadding", "style.padding-right": "this.rightPadding" }, classAttribute: "nb-tree-grid-cell" }, providers: [{ provide: NbCdkCell, useExisting: NbTreeGridCellDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridCellDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'td[nbTreeGridCell]',
                    host: {
                        'class': 'nb-tree-grid-cell',
                        'role': 'gridcell',
                    },
                    providers: [{ provide: NbCdkCell, useExisting: NbTreeGridCellDirective }],
                }]
        }], ctorParameters: function () { return [{ type: i1.NbTreeGridColumnDefDirective }, { type: i0.ElementRef }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [NB_TREE_GRID]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [NB_WINDOW]
                }] }, { type: i2.DomSanitizer }, { type: i3.NbLayoutDirectionService }, { type: i4.NbColumnsService }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { columnWidth: [{
                type: HostBinding,
                args: ['style.width']
            }], leftPadding: [{
                type: HostBinding,
                args: ['style.padding-left']
            }], rightPadding: [{
                type: HostBinding,
                args: ['style.padding-right']
            }] } });
export class NbTreeGridHeaderCellDirective extends NbHeaderCellDirective {
    constructor(columnDef, elementRef, tree, columnService, cd) {
        super(columnDef, elementRef);
        this.columnService = columnService;
        this.cd = cd;
        this.destroy$ = new Subject();
        this.tree = tree;
    }
    get columnWidth() {
        this.latestWidth = this.tree.getColumnWidth();
        return this.latestWidth || null;
    }
    ngOnInit() {
        this.columnService.onColumnsChange()
            .pipe(filter(() => this.latestWidth !== this.tree.getColumnWidth()), takeUntil(this.destroy$))
            .subscribe(() => this.cd.detectChanges());
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
}
NbTreeGridHeaderCellDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridHeaderCellDirective, deps: [{ token: i1.NbTreeGridColumnDefDirective }, { token: i0.ElementRef }, { token: NB_TREE_GRID }, { token: i4.NbColumnsService }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Directive });
NbTreeGridHeaderCellDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbTreeGridHeaderCellDirective, selector: "th[nbTreeGridHeaderCell]", host: { attributes: { "role": "columnheader" }, properties: { "style.width": "this.columnWidth" }, classAttribute: "nb-tree-grid-header-cell" }, providers: [{ provide: NbCdkHeaderCell, useExisting: NbTreeGridHeaderCellDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridHeaderCellDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'th[nbTreeGridHeaderCell]',
                    host: {
                        'class': 'nb-tree-grid-header-cell',
                        'role': 'columnheader',
                    },
                    providers: [{ provide: NbCdkHeaderCell, useExisting: NbTreeGridHeaderCellDirective }],
                }]
        }], ctorParameters: function () { return [{ type: i1.NbTreeGridColumnDefDirective }, { type: i0.ElementRef }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [NB_TREE_GRID]
                }] }, { type: i4.NbColumnsService }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { columnWidth: [{
                type: HostBinding,
                args: ['style.width']
            }] } });
export class NbTreeGridFooterCellDirective extends NbFooterCellDirective {
    constructor(columnDef, elementRef, tree, columnService, cd) {
        super(columnDef, elementRef);
        this.columnService = columnService;
        this.cd = cd;
        this.destroy$ = new Subject();
        this.tree = tree;
    }
    get columnWidth() {
        this.latestWidth = this.tree.getColumnWidth();
        return this.latestWidth || null;
    }
    ngOnInit() {
        this.columnService.onColumnsChange()
            .pipe(filter(() => this.latestWidth !== this.tree.getColumnWidth()), takeUntil(this.destroy$))
            .subscribe(() => this.cd.detectChanges());
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
}
NbTreeGridFooterCellDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridFooterCellDirective, deps: [{ token: i1.NbTreeGridColumnDefDirective }, { token: i0.ElementRef }, { token: NB_TREE_GRID }, { token: i4.NbColumnsService }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Directive });
NbTreeGridFooterCellDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbTreeGridFooterCellDirective, selector: "td[nbTreeGridFooterCell]", host: { attributes: { "role": "gridcell" }, properties: { "style.width": "this.columnWidth" }, classAttribute: "nb-tree-grid-footer-cell" }, providers: [{ provide: NbCdkFooterCell, useExisting: NbTreeGridFooterCellDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTreeGridFooterCellDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'td[nbTreeGridFooterCell]',
                    host: {
                        'class': 'nb-tree-grid-footer-cell',
                        'role': 'gridcell',
                    },
                    providers: [{ provide: NbCdkFooterCell, useExisting: NbTreeGridFooterCellDirective }],
                }]
        }], ctorParameters: function () { return [{ type: i1.NbTreeGridColumnDefDirective }, { type: i0.ElementRef }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [NB_TREE_GRID]
                }] }, { type: i4.NbColumnsService }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { columnWidth: [{
                type: HostBinding,
                args: ['style.width']
            }] } });
