// @nebular/theme/esm2020/components/tree-grid/tree-grid-filter.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { Directive, HostListener, Input } from '@angular/core';
import { Subject } from 'rxjs';
import { debounceTime, takeUntil } from 'rxjs/operators';
import * as i0 from "@angular/core";
export class NbFilterDirective {
    filter(filterRequest) {
        this.filterable.filter(filterRequest);
    }
}
NbFilterDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFilterDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
NbFilterDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbFilterDirective, selector: "[nbFilter]", inputs: { filterable: ["nbFilter", "filterable"] }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFilterDirective, decorators: [{
            type: Directive,
            args: [{ selector: '[nbFilter]' }]
        }], propDecorators: { filterable: [{
                type: Input,
                args: ['nbFilter']
            }] } });
/**
 * Helper directive to trigger data source's filter method when user types in input
 */
export class NbFilterInputDirective extends NbFilterDirective {
    constructor() {
        super(...arguments);
        this.search$ = new Subject();
        this.destroy$ = new Subject();
        /**
         * Debounce time before triggering filter method. Set in milliseconds.
         * Default 200.
         */
        this.debounceTime = 200;
    }
    ngOnInit() {
        this.search$
            .pipe(debounceTime(this.debounceTime), takeUntil(this.destroy$))
            .subscribe((query) => {
            super.filter(query);
        });
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
        this.search$.complete();
    }
    filter(event) {
        this.search$.next(event.target.value);
    }
}
NbFilterInputDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFilterInputDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NbFilterInputDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbFilterInputDirective, selector: "[nbFilterInput]", inputs: { filterable: ["nbFilterInput", "filterable"], debounceTime: "debounceTime" }, host: { listeners: { "input": "filter($event)" } }, providers: [{ provide: NbFilterDirective, useExisting: NbFilterInputDirective }], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbFilterInputDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[nbFilterInput]',
                    providers: [{ provide: NbFilterDirective, useExisting: NbFilterInputDirective }],
                }]
        }], propDecorators: { filterable: [{
                type: Input,
                args: ['nbFilterInput']
            }], debounceTime: [{
                type: Input
            }], filter: [{
                type: HostListener,
                args: ['input', ['$event']]
            }] } });
