// @nebular/theme/esm2020/components/stepper/stepper-button.directive.mjs
import { Directive, HostBinding, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./stepper.component";
export class NbStepperNextDirective {
    constructor(stepper) {
        this.stepper = stepper;
        this.type = 'submit';
    }
    onClick() {
        this.stepper.next();
    }
}
NbStepperNextDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbStepperNextDirective, deps: [{ token: i1.NbStepperComponent }], target: i0.ɵɵFactoryTarget.Directive });
NbStepperNextDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbStepperNextDirective, selector: "button[nbStepperNext]", inputs: { type: "type" }, host: { listeners: { "click": "onClick()" }, properties: { "attr.type": "this.type" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbStepperNextDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'button[nbStepperNext]',
                }]
        }], ctorParameters: function () { return [{ type: i1.NbStepperComponent }]; }, propDecorators: { type: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['attr.type']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
export class NbStepperPreviousDirective {
    constructor(stepper) {
        this.stepper = stepper;
        this.type = 'button';
    }
    onClick() {
        this.stepper.previous();
    }
}
NbStepperPreviousDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbStepperPreviousDirective, deps: [{ token: i1.NbStepperComponent }], target: i0.ɵɵFactoryTarget.Directive });
NbStepperPreviousDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.0", type: NbStepperPreviousDirective, selector: "button[nbStepperPrevious]", inputs: { type: "type" }, host: { listeners: { "click": "onClick()" }, properties: { "attr.type": "this.type" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbStepperPreviousDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'button[nbStepperPrevious]',
                }]
        }], ctorParameters: function () { return [{ type: i1.NbStepperComponent }]; }, propDecorators: { type: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['attr.type']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
