// @nebular/theme/esm2020/components/radio/radio.module.mjs
/*
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbRadioComponent } from './radio.component';
import { NbRadioGroupComponent } from './radio-group.component';
import * as i0 from "@angular/core";
export class NbRadioModule {
}
NbRadioModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRadioModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbRadioModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRadioModule, declarations: [NbRadioComponent, NbRadioGroupComponent], exports: [NbRadioComponent, NbRadioGroupComponent] });
NbRadioModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRadioModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRadioModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    exports: [NbRadioComponent, NbRadioGroupComponent],
                    declarations: [NbRadioComponent, NbRadioGroupComponent],
                }]
        }] });
