// @nebular/theme/esm2020/components/route-tabset/route-tabset.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbRouteTabsetComponent } from './route-tabset.component';
import { NbMergeConfigsPipe } from './merge-configs.pipe';
import { NbIconModule } from '../icon/icon.module';
import * as i0 from "@angular/core";
export class NbRouteTabsetModule {
}
NbRouteTabsetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRouteTabsetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbRouteTabsetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRouteTabsetModule, declarations: [NbRouteTabsetComponent, NbMergeConfigsPipe], imports: [NbSharedModule, NbIconModule], exports: [NbRouteTabsetComponent] });
NbRouteTabsetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRouteTabsetModule, imports: [[NbSharedModule, NbIconModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbRouteTabsetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [NbSharedModule, NbIconModule],
                    declarations: [NbRouteTabsetComponent, NbMergeConfigsPipe],
                    exports: [NbRouteTabsetComponent],
                }]
        }] });
