// @nebular/theme/esm2020/components/tag/tag.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NbIconModule } from '../icon/icon.module';
import { NbTagComponent } from './tag.component';
import { NbTagListComponent } from './tag-list.component';
import { NbTagInputDirective } from './tag-input.directive';
import * as i0 from "@angular/core";
export class NbTagModule {
}
NbTagModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTagModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbTagModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTagModule, declarations: [NbTagComponent,
        NbTagListComponent,
        NbTagInputDirective], imports: [CommonModule,
        NbIconModule], exports: [NbTagComponent,
        NbTagListComponent,
        NbTagInputDirective] });
NbTagModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTagModule, imports: [[
            CommonModule,
            NbIconModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbTagModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        NbIconModule,
                    ],
                    declarations: [
                        NbTagComponent,
                        NbTagListComponent,
                        NbTagInputDirective,
                    ],
                    exports: [
                        NbTagComponent,
                        NbTagListComponent,
                        NbTagInputDirective,
                    ],
                }]
        }] });
