// @nebular/theme/esm2020/components/input/input.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbInputDirective } from './input.directive';
import * as i0 from "@angular/core";
const NB_INPUT_COMPONENTS = [
    NbInputDirective,
];
export class NbInputModule {
}
NbInputModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbInputModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbInputModule, declarations: [NbInputDirective], imports: [NbSharedModule], exports: [NbInputDirective] });
NbInputModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbInputModule, imports: [[NbSharedModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbInputModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [NbSharedModule],
                    declarations: NB_INPUT_COMPONENTS,
                    exports: NB_INPUT_COMPONENTS,
                }]
        }] });
