// @nebular/theme/esm2020/components/alert/alert.module.mjs
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { NgModule } from '@angular/core';
import { NbSharedModule } from '../shared/shared.module';
import { NbAlertComponent } from './alert.component';
import * as i0 from "@angular/core";
export class NbAlertModule {
}
NbAlertModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAlertModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NbAlertModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAlertModule, declarations: [NbAlertComponent], imports: [NbSharedModule], exports: [NbAlertComponent] });
NbAlertModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAlertModule, imports: [[
            NbSharedModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.0", ngImport: i0, type: NbAlertModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        NbSharedModule,
                    ],
                    declarations: [
                        NbAlertComponent,
                    ],
                    exports: [
                        NbAlertComponent,
                    ],
                }]
        }] });
